splice/web/api/tile.py

```diff
@@ -0,0 +1,121 @@
+                                   help='base64 encoded roll over image', location='json')
+        self.reqparse.add_argument('enhanced_image_uri', type=str, required=True,
+                                   help='base64 encoded static image', location='json')
+        self.reqparse.add_argument('bg_color', type=str, default="",
+                                   help='Background color', location='json')
+        self.reqparse.add_argument('explanation', type=unicode, default="",
+                                   help='Contextual explanation', location='json')
+        super(TileListAPI, self).__init__()
+
+    def get(self, adgroup_id):
+        tiles = get_tiles_by_adgroup_id(adgroup_id)
+        if len(tiles) == 0:
+            return {"message": "No tiles found"}, 404
+        else:
+            return {"message": marshal(tiles, tile_fields)}
+
+    def post(self, adgroup_id):
+        """ HTTP end point to create new tile. Note the initial status of a new
+        tile is always set as 'unapproved'
+        """
+        args = self.reqparse.parse_args()
+        # the status of new tiles are alway set as unapproved
+        args["status"] = "unapproved"
+
+        try:
+            # validate background color, url, and image fields again
+            jsonschema.validate(args, API_TILE_SCHEMA)
+            with session_scope() as session:
+                new = insert_tile(session, args)
+        except jsonschema.exceptions.ValidationError as e:
+            return {"message": e.message}, 400
+        except InvalidRequestError as e:
+            return {"message": e.message}, 400
+        else:
+            return {"message": marshal(new, tile_fields)}, 201
+
+
+class TileAPI(Resource):
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('adgroup_id', type=int, required=True,
+                                   help='adgroup ID', location='json')
+        self.reqparse.add_argument('status', type=str, required=True,
+                                   choices=Tile.STATUS,
+                                   help='Status', location='json')
+        super(TileAPI, self).__init__()
+
+    def get(self, adgroup_id, tile_id):
+        tile = get_tile(tile_id)
+        if tile is None:
+            return {"message": "No tile found"}, 404
+        else:
+            return {"message": marshal(tile, tile_fields)}
+
+    def put(self, adgroup_id, tile_id):
+        args = self.reqparse.parse_args()
+        try:
+            with session_scope() as session:
+                tile = update_tile(session, tile_id, args)
+        except NoResultFound as e:
+            return {"message": e.message}, 404
+        else:
+            return {"message": marshal(tile, tile_fields)}, 200
+
+
+api.add_resource(TileListAPI, '/adgroups/<int:adgroup_id>/tiles', endpoint='tiles')
+api.add_resource(TileAPI, '/adgroups/<int:adgroup_id>/tile/<int:tile_id>', endpoint='tile')
```

Comment:
see, I find it weird that if you know the tile_id then why do you also have to pass the adgroup_id? Why is it required and not the campaign and account as well then?